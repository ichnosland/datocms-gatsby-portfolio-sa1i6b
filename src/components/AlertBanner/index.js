/**
*
* AlertBanner
*
*/

import styled, { css } from 'styled-components';
import { lighten, darken } from 'polished';
import FlexBox from '../FlexBox';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BannerStyle = css`
  ${({ tag }) => tag && css`display: inline-flex;`};
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  text-align: center;
  margin: ${(props) => props.margin ? props.margin : '0'};
  border: ${(props) => props.border ? `1px solid ${props.theme.brand}` : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  ${({ actioncolor }) => actioncolor && css`
    border: 1px solid ${colore.actions[actioncolor]};
    background-color: ${lighten(0.4, colore.actions[actioncolor])};
    color: ${darken(0.2, colore.actions[actioncolor])};
    svg {
      fill: ${darken(0.2, colore.actions[actioncolor])};
    }
  `}
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const AlertBanner = styled(FlexBox)`
  ${BannerStyle}
  padding: ${(props) => props.padding ? props.padding : '0.6em  1.6em'};
  svg {
    fill: ${({ color, actioncolor }) => {
    if (color) return color;
    if (actioncolor) return actioncolor;

    return colore.ui.txt;
  }};
  }
`;

AlertBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const InfoBanner = styled(AlertBanner)`
  padding: ${(props) => props.padding ? props.padding : '0.6em  1em'};
  ${media.lt667`
    flex-direction: column;
    svg {
      margin: 10px 0;
    }
  `}
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '0.6em'};
  `}
`;

InfoBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const IconBanner = styled(FlexBox)`
  ${BannerStyle}
  align-self: stretch;
  align-items: center;
  padding: ${(props) => props.padding ? props.padding : '9px 16px'};
  div {
    margin-right: 12px;
    align-items: center;
    &:last-child {
      margin-right: 0;
    }
  }
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '7px 12px'};
  `}
`;

IconBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export default AlertBanner;
