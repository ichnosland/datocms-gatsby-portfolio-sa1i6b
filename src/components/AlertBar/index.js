/**
*
* AlertBar
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import styled, { css } from 'styled-components';

import Div from '../Div';
import FlexBox from '../FlexBox';
import Svg from '../Svg';
import { IconBtn } from '../TopBar';
import { H3 } from '../Heading';
import media from 'style/mediainjector';
import { colore } from 'style/color';
import icon from 'icons/globals';
import { ContainerStyle } from '../Container';

export const AlertWrap = styled(Div)`
  position: absolute;
  top: 0;
  left: 0;
  color: ${colore.ui.contrast};
  width: 100%;
  height: 56px;
  padding: 0 20px;
  background-color: ${(props) => props.theme.brand};
  ${({ giusto }) => giusto && css`background-color: ${colore.actions.okay};`}
  ${({ sbagliato }) => sbagliato && css`background-color: ${colore.actions.error};`}
  ${({ aiuto }) => aiuto && css`background-color: ${colore.actions.help};`}
  z-index: 8;
  transition: all 0.5s;
  ${media.desktop`
    border-top-left-radius: ${(props) => props.versione ? '3px' : '20px'};
    border-top-right-radius: ${(props) => props.versione ? '3px' : '20px'};
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

AlertWrap.defaulProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const BarContainer = styled(FlexBox)`
  ${ContainerStyle}
  height: 100%;
  ${(props) => props.helpFx && css`
    justify-content: space-between;
    &::after {
      content: "";
      display: inline;
    }
  `}
  h3 {
    color: ${colore.ui.contrast};
  }
`;

class AlertBar extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      giusto,
      sbagliato,
      aiuto,
      versione,
      show,
      helpFx,
    } = this.props;
    return (
      <AlertWrap
        giusto={giusto}
        sbagliato={sbagliato}
        aiuto={aiuto}
        show={show}
        versione={versione}
      >
        <BarContainer helpFx={helpFx}>
          {helpFx &&
            <IconBtn onClick={helpFx}>
              <Svg {...icon.topBarFeedback} />
            </IconBtn>
          }
          <H3>
            {giusto &&
              'Risposta corretta!'
            }
            {sbagliato &&
              'Hai sbagliato!'
            }
            {aiuto &&
              'Aiuto'
            }
          </H3>
        </BarContainer>
      </AlertWrap>
    );
  }
}

AlertBar.propTypes = {
  giusto: PropTypes.bool,
  sbagliato: PropTypes.bool,
  aiuto: PropTypes.bool,
  versione: PropTypes.bool,
  show: PropTypes.bool.isRequired,
  helpFx: PropTypes.func,
};

export default AlertBar;
