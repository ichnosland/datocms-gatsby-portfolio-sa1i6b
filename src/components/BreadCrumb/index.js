/**
*
* BreadCrumb
*
*/

import styled, { css } from 'styled-components';

import Div from 'components/Div';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BreadCrumb = styled(Div)`
  color: ${(props) => props.color};
  position: relative;
  display: ${(props) => props.display ? props.display : 'inline-block'};
  text-align: center;
  border: ${(props) => props.border ? props.border : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const Crumb = styled(Div)`
  position: relative;
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
  flex-wrap: nowrap;
  padding: 0.4em 1.4em 0.4em 0.8em;
  &:first-child {
    padding-left: 1.2em;
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:not(:first-child):not(:last-child) {
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:only-child {
    &:after {
      content: "";
      position: relative;
    }
  }
`;

BreadCrumb.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export default BreadCrumb;
