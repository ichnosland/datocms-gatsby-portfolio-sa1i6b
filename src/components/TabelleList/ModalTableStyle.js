/**
*
* ModalTableStyle
*
*/

import { transparentize } from 'polished';
import { css } from 'styled-components';

import { colore } from 'style/color';

export const ModalTableStyle = css`
  color: ${colore.ui.txt};
  text-align: left;
  border-spacing: 0;
  border: none;
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  user-select: ${(props) => props.userSelect};
  th {
    color: ${colore.ui.contrast};
    font-weight: 400;
    padding: 9px 12px;
    background-color: ${(props) => props.theme.brand};
  }
  td {
    padding: 9px 12px;
    overflow:hidden;
    word-break:normal;
    border-bottom-width: 1px;
    border-bottom-style: solid;
    border-bottom-color: ${(props) => props.theme.brand && transparentize(0.8, props.theme.brand)};
    &:nth-child(odd) {
      width: 6rem;
      color: ${(props) => props.theme.brand};
      background-color:  ${(props) => props.theme.brand && transparentize(0.9, props.theme.brand)};
      border-right-width: 1px;
      border-right-style: solid;
      border-right-color:  ${(props) => props.theme.brand && transparentize(0.8, props.theme.brand)};
      text-align: center;
    }
    &:nth-child(even) {
      background: ${colore.ui.contrast};
    }
  }
  tr {
    &:last-child {
      td{
        &:first-child {
          border-bottom: none;
        }
        &:last-child {
          border-bottom: none;
        }
      }
    }
  }
`;

ModalTableStyle.DefaultProps = {
  theme: {
    brand: 'mediumseagreen',
  },
};
