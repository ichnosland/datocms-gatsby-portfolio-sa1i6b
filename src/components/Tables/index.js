/*
 *
 * Tables
 *
 */

import styled, { css } from 'styled-components';

import media from 'style/mediainjector';
import { colore } from 'style/color';

export const Table = styled.table`
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  border-width: 1px;
  border-style: solid;
  background-color: ${(props) => props.bgcolor};
  ${media.lt480`
    font-size: 12px;
  `}
  ${({ fixed }) => fixed && css`
    width: 100%;
    table-layout: fixed;
  `}
  ${({ pricing }) => pricing && css`
    color: white;
    border-color: white;
  `}
  ${({ neutral }) => neutral && css`
    border-color: ${colore.landing.dark};
    td {
      vertical-align: top;
      padding: 16px;
      ${media.lt667`
        padding: 12px;
        ul {
          padding: 10px;
        }
      `}
    }
  `}
  strong {
    font-weight: 700;
  }
`;

export const Tr = styled.tr`
  border-top: ${(props) => props.Bt ? props.Bt : 'none'};
  border-right: ${(props) => props.Br ? props.Br : 'none'};
  border-left: ${(props) => props.Bl ? props.Bl : 'none'};
  border-bottom: ${(props) => props.Bb ? props.Bb : 'none'};
  ${({ fullBg }) => fullBg && css`
    background-color: ${(props) => props.Bg ? props.Bg : 'white'};
    color: ${(props) => props.color ? props.color : '#0085b2'};
  `}
`;

export const Td = styled.td`
  padding: 14px 5px;
  overflow:hidden;
  word-break:normal;
  background-color: ${(props) => props.bgcolor};
  border-top: ${(props) => props.Bt ? props.Bt : 'none'};
  border-right: ${(props) => props.Br ? props.Br : 'none'};
  border-left: ${(props) => props.Bl ? props.Bl : 'none'};
  border-bottom: ${(props) => props.Bb ? props.Bb : 'none'};
`;

export const NoteTd = styled(Td)`
  padding: 6px 5px;
  font-size: 14px;
`;
