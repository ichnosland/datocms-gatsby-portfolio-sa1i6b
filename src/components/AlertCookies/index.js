/**
*
* AlertCookies
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import P from 'components/Paragraph';
import { Button } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const AlertCookiesWrap = styled(Div)`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 0 20px 20px;
  z-index: 1;
  ${media.lt667`
    padding: 0 10px 30px;
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

export const CookiesBanner = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  max-width: 860px;
  margin: 0 auto;
  font-size: 14px;
  border: ${(props) => props.border};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : 'rgba(255,255,255,.95)'};
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,.15);
  box-shadow: 0 2px 1px rgba(0,0,0,.25);
  p {
    padding: 20px;
    margin: 0;
    &:last-child {
      padding: 0 20px 10px;
      text-align: right;
      ${media.lt480`
        padding: 0 20px 20px;
        text-align: center;
      `}
    }
  }
  ${media.lt667`
    font-size: 12px;
  `}
`;

class AlertCookies extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { show, color, border, bgcolor, onClickFunction } = this.props;
    return (
      <AlertCookiesWrap show={show}>
        <CookiesBanner
          color={color}
          border={border}
          bgcolor={bgcolor}
        >
          <P>Al fine di offrire il miglior servizio possibile questo sito utilizza cookies tecnici e di terze parti per funzionalità quali l&apos;accesso tramite social network e/o la visualizzazione di media. Se non acconsenti all&apos;utilizzo dei cookie di terze parti, alcune di queste funzionalità potrebbero essere non disponibili. Chiudendo questo banner o proseguendo con la navigazione acconsentirai al loro impiego. Per saperne di pi&ugrave; o negare il consenso a tutti o ad alcuni cookie puoi consultare <a href="https://maieuticallabs.it/privacy" target="_blank"><span>l&apos;informativa sulla privacy</span></a>.</P>
          <P><Button actioncolor="hint" standard={1} onClick={onClickFunction}>Chiudi</Button></P>
        </CookiesBanner>
      </AlertCookiesWrap>
    );
  }
}

AlertCookies.propTypes = {
  color: PropTypes.string,
  border: PropTypes.string,
  bgcolor: PropTypes.string,
  show: PropTypes.bool.isRequired,
  onClickFunction: PropTypes.func,
};

export default AlertCookies;
