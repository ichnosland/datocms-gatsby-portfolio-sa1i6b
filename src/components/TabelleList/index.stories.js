import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import { ThemeProvider } from 'styled-components';
import 'global-styles';
import { allThemes, themeOptions } from 'style/theme';
import TabelleList from './index';

const tabelle = {
  tabelle: [{
    intestazione: 'intestazione tabella uno',
    righe: [
      [{
        titolo: 'titolo riga 1 colonna 1',
      }, {
        titolo: 'titolo riga 1 colonna 2',
      }],
      [{
        titolo: 'titolo riga 2 colonna 1',
      }, {
        titolo: 'titolo riga 2 colonna 2',
      }],
    ],
  }, {
    intestazione: 'intestazione tabella due',
    righe: [
      [{
        titolo: 'titolo riga 1 colonna 1',
      }, {
        titolo: 'titolo riga 1 colonna 2',
      }],
    ],
  }],
};

storiesOf('TabelleList', module)
  .addDecorator(withKnobs)
  .add('TabelleList', (() => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <TabelleList
        {...tabelle}
        margin={text('margin', '0 0 16px')}
        radius={text('radius', '6px')}
      />
    </ThemeProvider>
  )));
