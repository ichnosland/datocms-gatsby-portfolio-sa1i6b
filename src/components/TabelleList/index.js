/**
*
* TabelleList
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Div from '../Div';
import { Table } from '../Tables';
import { ModalTableStyle } from './ModalTableStyle';

export const ModalTable = styled(Table)`
  ${ModalTableStyle}
`;

class TabelleList extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { tabelle, radius, margin, className, userSelect } = this.props;

    return ([
      tabelle.map((tabella, key) => (
        <Div
          radius={radius}
          overflow="hidden"
          margin={margin}
          key={`wrap_tabella_${key + 1}`}
          className={className}
        >
          <ModalTable
            key={`tabella_${key + 1}`}
            userSelect={userSelect}
          >
            {tabella.intestazione && (
              <thead>
                <tr>
                  <th colSpan={tabella.righe[0].length}>
                    {tabella.intestazione}
                  </th>
                </tr>
              </thead>
            )}
            {tabella.righe && (
              <tbody>
                {tabella.righe.map((riga, rigaKey) => (
                  <tr key={`tabella_${key + 1}_Tr_${rigaKey + 1}`}>
                    {riga.map((colonna, colTdKey) => (
                      <td
                        key={`tabella_Tr_${key + 1}_td_${colTdKey + 1}`}
                        dangerouslySetInnerHTML={{ __html: colonna.titolo }} // eslint-disable-line
                      />
                    ))}
                  </tr>
                ))}
              </tbody>
            )}
          </ModalTable>
        </Div>
      )),
    ]);
  }
}

TabelleList.propTypes = {
  tabelle: PropTypes.arrayOf(PropTypes.shape({
    intestazione: PropTypes.sTring,
    righe: PropTypes.arrayOf(
      PropTypes.arrayOf(PropTypes.shape({
        titolo: PropTypes.oneOfType([
          PropTypes.string.isRequired,
          PropTypes.number.isRequired,
        ]),
      }))
    ),
  })).isRequired,
  radius: PropTypes.string,
  margin: PropTypes.string,
  userSelect: PropTypes.string,
  className: PropTypes.string,
};

export default TabelleList;
