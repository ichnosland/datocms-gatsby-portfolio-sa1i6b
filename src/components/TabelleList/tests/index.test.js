import React from 'react';
import { shallow } from 'enzyme';

import { Table } from 'components/Tables';
import TabelleList, { ModalTable } from '../index';

const theme = {
  brand: '#00abe5',
};

const props = {
  tabelle: [{
    intestazione: 'intestazione tabella uno',
    righe: [
      [{
        titolo: 'titolo riga 1 colonna 1',
      }, {
        titolo: 'titolo riga 1 colonna 2',
      }],
    ],
  }, {
    intestazione: 'intestazione tabella due',
    righe: [
      [{
        titolo: 'titolo riga 1 colonna 1',
      }, {
        titolo: 'titolo riga 1 colonna 2',
      }],
    ],
  }],
};

describe('<TabelleList />', () => {
  it('dichiaro una tabella completa', () => {
    const renderedComponent = shallow(
      <TabelleList {...props} />
    );
    expect(renderedComponent.find(ModalTable).length).toBe(2);
    expect(renderedComponent.find('thead').length).toBe(2);
    expect(renderedComponent.find('tbody').length).toBe(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('dichiaro una tabella senza intestazione', () => {
    const mockProp = {
      ...props,
      tabelle: [{
        ...props.tabelle[0],
        intestazione: undefined,
      }],
    };
    const renderedComponent = shallow(
      <TabelleList {...mockProp} />
    );
    expect(renderedComponent.find('TabelleList__ModalTable').length).toBe(1);
    expect(renderedComponent.find('thead').length).toBe(0);
    expect(renderedComponent.find('tbody').length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props margin', () => {
    const margin = '30px auto';
    const renderedComponent = shallow(<TabelleList {...props} margin={margin} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<ModalTable />', () => {
  it('should render a Table', () => {
    const renderedComponent = shallow(<ModalTable />);
    expect(renderedComponent.find(Table).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props theme', () => {
    const renderedComponent = shallow(<ModalTable theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props bgcolor', () => {
    const bgcolor = 'red';
    const renderedComponent = shallow(<ModalTable bgcolor={bgcolor} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props userSelect', () => {
    const userSelect = 'auto';
    const renderedComponent = shallow(<ModalTable userSelect={userSelect} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
