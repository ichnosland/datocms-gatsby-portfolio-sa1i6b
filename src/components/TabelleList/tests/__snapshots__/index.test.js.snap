// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalTable /> should render a Table 1`] = `
.c0 {
  color: rgb(78,78,78);
  text-align: left;
  border-spacing: 0;
  border: none;
  background-color: rgb(255,255,255);
}

.c0 th {
  color: rgb(255,255,255);
  font-weight: 400;
  padding: 9px 12px;
}

.c0 td {
  padding: 9px 12px;
  overflow: hidden;
  word-break: normal;
  border-bottom-width: 1px;
  border-bottom-style: solid;
}

.c0 td:nth-child(odd) {
  width: 6rem;
  border-right-width: 1px;
  border-right-style: solid;
  text-align: center;
}

.c0 td:nth-child(even) {
  background: rgb(255,255,255);
}

.c0 tr:last-child td:first-child {
  border-bottom: none;
}

.c0 tr:last-child td:last-child {
  border-bottom: none;
}

<Tables__Table
  className="c0"
/>
`;

exports[`<ModalTable /> should render its css w/ props bgcolor 1`] = `
.c0 {
  color: rgb(78,78,78);
  text-align: left;
  border-spacing: 0;
  border: none;
  background-color: red;
}

.c0 th {
  color: rgb(255,255,255);
  font-weight: 400;
  padding: 9px 12px;
}

.c0 td {
  padding: 9px 12px;
  overflow: hidden;
  word-break: normal;
  border-bottom-width: 1px;
  border-bottom-style: solid;
}

.c0 td:nth-child(odd) {
  width: 6rem;
  border-right-width: 1px;
  border-right-style: solid;
  text-align: center;
}

.c0 td:nth-child(even) {
  background: rgb(255,255,255);
}

.c0 tr:last-child td:first-child {
  border-bottom: none;
}

.c0 tr:last-child td:last-child {
  border-bottom: none;
}

<Tables__Table
  bgcolor="red"
  className="c0"
/>
`;

exports[`<ModalTable /> should render its css w/ props theme 1`] = `
.c0 {
  color: rgb(78,78,78);
  text-align: left;
  border-spacing: 0;
  border: none;
  background-color: rgb(255,255,255);
}

.c0 th {
  color: rgb(255,255,255);
  font-weight: 400;
  padding: 9px 12px;
  background-color: #00abe5;
}

.c0 td {
  padding: 9px 12px;
  overflow: hidden;
  word-break: normal;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: rgba(0,171,229,0.2);
}

.c0 td:nth-child(odd) {
  width: 6rem;
  color: #00abe5;
  background-color: rgba(0,171,229,0.1);
  border-right-width: 1px;
  border-right-style: solid;
  border-right-color: rgba(0,171,229,0.2);
  text-align: center;
}

.c0 td:nth-child(even) {
  background: rgb(255,255,255);
}

.c0 tr:last-child td:first-child {
  border-bottom: none;
}

.c0 tr:last-child td:last-child {
  border-bottom: none;
}

<Tables__Table
  className="c0"
  theme={
    Object {
      "brand": "#00abe5",
    }
  }
/>
`;

exports[`<ModalTable /> should render its css w/ props userSelect 1`] = `
.c0 {
  color: rgb(78,78,78);
  text-align: left;
  border-spacing: 0;
  border: none;
  background-color: rgb(255,255,255);
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
}

.c0 th {
  color: rgb(255,255,255);
  font-weight: 400;
  padding: 9px 12px;
}

.c0 td {
  padding: 9px 12px;
  overflow: hidden;
  word-break: normal;
  border-bottom-width: 1px;
  border-bottom-style: solid;
}

.c0 td:nth-child(odd) {
  width: 6rem;
  border-right-width: 1px;
  border-right-style: solid;
  text-align: center;
}

.c0 td:nth-child(even) {
  background: rgb(255,255,255);
}

.c0 tr:last-child td:first-child {
  border-bottom: none;
}

.c0 tr:last-child td:last-child {
  border-bottom: none;
}

<Tables__Table
  className="c0"
  userSelect="auto"
/>
`;

exports[`<TabelleList /> dichiaro una tabella completa 1`] = `
Array [
  <Div
    key="wrap_tabella_1"
    overflow="hidden"
  >
    <TabelleList__ModalTable
      key="tabella_1"
    >
      <thead>
        <tr>
          <th
            colSpan={2}
          >
            intestazione tabella uno
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="tabella_1_Tr_1"
        >
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 1",
              }
            }
            key="tabella_Tr_1_td_1"
          />
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 2",
              }
            }
            key="tabella_Tr_1_td_2"
          />
        </tr>
      </tbody>
    </TabelleList__ModalTable>
  </Div>,
  <Div
    key="wrap_tabella_2"
    overflow="hidden"
  >
    <TabelleList__ModalTable
      key="tabella_2"
    >
      <thead>
        <tr>
          <th
            colSpan={2}
          >
            intestazione tabella due
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="tabella_2_Tr_1"
        >
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 1",
              }
            }
            key="tabella_Tr_2_td_1"
          />
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 2",
              }
            }
            key="tabella_Tr_2_td_2"
          />
        </tr>
      </tbody>
    </TabelleList__ModalTable>
  </Div>,
]
`;

exports[`<TabelleList /> dichiaro una tabella senza intestazione 1`] = `
<Div
  key="wrap_tabella_1"
  overflow="hidden"
>
  <TabelleList__ModalTable
    key="tabella_1"
  >
    <tbody>
      <tr
        key="tabella_1_Tr_1"
      >
        <td
          dangerouslySetInnerHTML={
            Object {
              "__html": "titolo riga 1 colonna 1",
            }
          }
          key="tabella_Tr_1_td_1"
        />
        <td
          dangerouslySetInnerHTML={
            Object {
              "__html": "titolo riga 1 colonna 2",
            }
          }
          key="tabella_Tr_1_td_2"
        />
      </tr>
    </tbody>
  </TabelleList__ModalTable>
</Div>
`;

exports[`<TabelleList /> should render its css w/ props margin 1`] = `
Array [
  <Div
    key="wrap_tabella_1"
    margin="30px auto"
    overflow="hidden"
  >
    <TabelleList__ModalTable
      key="tabella_1"
    >
      <thead>
        <tr>
          <th
            colSpan={2}
          >
            intestazione tabella uno
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="tabella_1_Tr_1"
        >
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 1",
              }
            }
            key="tabella_Tr_1_td_1"
          />
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 2",
              }
            }
            key="tabella_Tr_1_td_2"
          />
        </tr>
      </tbody>
    </TabelleList__ModalTable>
  </Div>,
  <Div
    key="wrap_tabella_2"
    margin="30px auto"
    overflow="hidden"
  >
    <TabelleList__ModalTable
      key="tabella_2"
    >
      <thead>
        <tr>
          <th
            colSpan={2}
          >
            intestazione tabella due
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="tabella_2_Tr_1"
        >
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 1",
              }
            }
            key="tabella_Tr_2_td_1"
          />
          <td
            dangerouslySetInnerHTML={
              Object {
                "__html": "titolo riga 1 colonna 2",
              }
            }
            key="tabella_Tr_2_td_2"
          />
        </tr>
      </tbody>
    </TabelleList__ModalTable>
  </Div>,
]
`;
